src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java

```diff
@@ -55,133 +54,147 @@
+                    subscriptionService.modifySubscription(subscription, subscriptionName);
+                    LOG.debug("Subscription updating is completed: " + subscriptionName);
+                } else {
+                    LOG.error("Subscription is not found: " + subscriptionName);
+                    errorMap.put(subscriptionName, "Subscription is not found");
+                }
+            } catch (Exception e) {
+                LOG.error("Error on subscription " + subscriptionName + ". " + e.getMessage());
+                errorMap.put(subscriptionName, e.getMessage());
+            }
+        });
+        if (errorMap.isEmpty()) {
+            return new ResponseEntity<>(HttpStatus.OK);
         } else {
-            LOG.error("Subscription :" + subscription.getSubscriptionName() + " can't be found.");
-            subscriptionResponse.setMsg("Subscription can't be found");
-            subscriptionResponse.setStatusCode(HttpStatus.BAD_REQUEST.value());
-            return new ResponseEntity<>(subscriptionResponse, HttpStatus.BAD_REQUEST);
+            return new ResponseEntity<>(getSubscriptionResponseList(errorMap), HttpStatus.BAD_REQUEST);
         }
     }
 
     @Override
     @CrossOrigin
-    @ApiOperation(value = "Removes the subscription from the database")
-    public ResponseEntity<SubscriptionResponse> deleteSubscriptionById(@PathVariable String subscriptionName) {
-        SubscriptionResponse subscriptionResponse = new SubscriptionResponse();
-        LOG.info("Subscription :" + subscriptionName + " delete started");
-        if (subscriptionService.deleteSubscription(subscriptionName)) {
-            LOG.info("Subscription :" + subscriptionName + " deleted Successfully");
-            subscriptionResponse.setMsg("Deleted Successfully");
-            subscriptionResponse.setStatusCode(HttpStatus.OK.value());
-            return new ResponseEntity<>(subscriptionResponse, HttpStatus.OK);
+    @ApiOperation(value = "Removes the subscriptions from the database")
+    public ResponseEntity<List<SubscriptionResponse>> deleteSubscriptionById(@PathVariable String subscriptionNames) {
+        errorMap = new HashMap<>();
+        // set is used to prevent subscription names repeating
+        Set<String> subscriptionNamesList = new HashSet<>(Arrays.asList(subscriptionNames.split(",")));
+
+        subscriptionNamesList.forEach(subscriptionName -> {
+            LOG.debug("Subscription deleting has been started: " + subscriptionName);
+            if (subscriptionService.deleteSubscription(subscriptionName)) {
+                LOG.debug("Subscription is deleted successfully: " + subscriptionName);
+            } else {
+                LOG.error("Subscription is not found :: " + subscriptionName);
+                errorMap.put(subscriptionName, "Subscription is not found");
+            }
+        });
+        if (errorMap.isEmpty()) {
+            return new ResponseEntity<>(HttpStatus.OK);
         } else {
```

Comment:
Maybe a minor comment. I guess else case is not needed here and always do the "return new Response......." .